### PR TITLE
refactor ChatOptions Builder

### LINKED ISSUE #1875: DefaultChatOptions should not be an inner class ChatOptionsBuilder, the build method should be inside the interface ChatOptions
This is similar to how we have ChatClient.

We can deprecate the top level ChatOptionsBuilder class and have the two sit side by side for one release cycle.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Deprecate existing ChatOptionsBuilder and its inner class DefaultChatOptions
 - Create a new builder interface ChatOptions.Builder for building the Chat options
  - Create an explicit DefaultChatOptions
  - Create DefaultChatOptionBuilder which can create DefaultChatOptions

Resolves #1875

### CODE REVIEW

File: spring-ai-core/src/main/java/org/springframework/ai/chat/prompt/ChatOptions.java [hunk 1]
```diff
@@ -89,4 +89,82 @@ public interface ChatOptions extends ModelOptions {
 	 */
 	ChatOptions copy();
 
+	/**
+	 * Creates a new {@link ChatOptions.Builder} to create the default
+	 * {@link ChatOptions}.
+	 * @return Returns a new {@link ChatOptions.Builder}.
+	 */
+	static ChatOptions.Builder builder() {
```
Comment (ilayaperumalg) [inline spring-ai-core/src/main/java/org/springframework/ai/chat/prompt/ChatOptions.java:97]: I think we need to have public access to this builder() for extensibility?

File: models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/AnthropicChatModel.java [hunk 2]
```diff
@@ -471,13 +470,13 @@ private List<AnthropicApi.Tool> getFunctionTools(Set<String> functionNames) {
 	}
 
 	private ChatOptions buildRequestOptions(AnthropicApi.ChatCompletionRequest request) {
-		return ChatOptionsBuilder.builder()
-			.withModel(request.model())
```
Comment (markpollack) [inline models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/AnthropicChatModel.java:475]: need to keep the 'with' methods for once cycle of deprecation.

File: spring-ai-core/src/main/java/org/springframework/ai/chat/prompt/ChatOptionsBuilder.java [hunk 3]
```diff
@@ -18,6 +18,7 @@
 
 import java.util.List;
 
+@Deprecated(forRemoval = true, since = "1.0.0-M5")
```
Comment (markpollack) [inline spring-ai-core/src/main/java/org/springframework/ai/chat/prompt/ChatOptionsBuilder.java:21]: need a @deprecated javadoc command to point users to the new implementation.

**Comment by markpollack** [comment]
merged in f252b2417a53db6be32480940a27159b1d45469c

**Pull request closed without merging**